src/DirectoryContents.cpp

```diff
@@ -0,0 +1,344 @@
+// � Copyright 2010 - 2019 BlackTopp Studios Inc.
+/* This file is part of The Mezzanine Engine.
+
+    The Mezzanine Engine is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    The Mezzanine Engine is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with The Mezzanine Engine.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/* The original authors have included a copy of the license specified above in the
+   'Docs' folder. See 'gpl.txt'
+*/
+/* We welcome the use of the Mezzanine engine to anyone, including companies who wish to
+   Build professional software and charge for their product.
+
+   However there are some practical restrictions, so if your project involves
+   any of the following you should contact us and we will try to work something
+   out:
+    - DRM or Copy Protection of any kind(except Copyrights)
+    - Software Patents You Do Not Wish to Freely License
+    - Any Kind of Linking to Non-GPL licensed Works
+    - Are Currently In Violation of Another Copyright Holder's GPL License
+    - If You want to change our code and not add a few hundred MB of stuff to
+        your distribution
+
+   These and other limitations could cause serious legal problems if you ignore
+   them, so it is best to simply contact us or the Free Software Foundation, if
+   you have any questions.
+
+   Joseph Toppi - toppij@gmail.com
+   John Blackwood - makoenergy02@gmail.com
+*/
+
+#include "CrossPlatformExport.h"
+
+#ifdef MEZZ_Windows
+    #define _WIN32_WINNT 0x0601
+#endif
+
+#include "DirectoryContents.h"
+#include "PathUtilities.h"
+#include "StringTools.h"
+
+#ifdef MEZZ_Windows
+    #define WIN32_LEAN_AND_MEAN
+
+    #include <Windows.h>
+#else
+    #include <stdio.h>
+    #include <sys/stat.h>
+    #include <sys/types.h>
+    #include <dirent.h>
+    #include <unistd.h>
+#endif
+
+#include "PlatformUndefs.h"
+
+namespace
+{
+    using namespace Mezzanine;
+
+#ifdef MEZZ_Windows
+    /// @brief Convenience type because Windows likes fatty Strings.
+    using WideString = std::wstring;
+
+    /// @brief Demotes a size_t to an int, which is required by the Win API.
+    /// @brief ToDemote The large integral type to be demoted.
+    /// @return Returns an int of the same value if it can be stored, or asserts otherwise.
+    [[nodiscard]]
+    int IntDemote(const size_t ToDemote)
```

Comment:
This has more general purpose use and might be put in Foundation or StaticFoundation.

Reply:
Issue for this has been created. #4